homeassistant/components/bluetooth/manager.py

```diff
@@ -281,24 +279,13 @@ def scanner_adv_received(self, service_info: BluetoothServiceInfoBleak) -> None:
             matched_domains,
         )
 
-        if (
-            not matched_domains
-            and not self._callbacks
-            and not self._connectable_callbacks
-        ):
-            return
-
-        for connectable_callback in (True, False):
-            for callback, matcher in self._get_callbacks_by_type(connectable_callback):
-                if matcher and not ble_device_matches(matcher, service_info):
-                    continue
-                try:
-                    callback(service_info, BluetoothChange.ADVERTISEMENT)
-                except Exception:  # pylint: disable=broad-except
-                    _LOGGER.exception("Error in bluetooth callback")
+        for match in self._callback_index.match_callbacks(service_info):
+            callback = match[CALLBACK]
+            try:
+                callback(service_info, BluetoothChange.ADVERTISEMENT)
+            except Exception:  # pylint: disable=broad-except
```

Comment:
Can we make this less broad?

Reply:
The callback is coming from outside the integration so If one integration registers a callback and it throws and we don't have a broad catch ere all the other ones that match after it would not get run.